src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/LockFileUtils.cs

```diff
@@ -685,6 +731,19 @@ private static void AddDependencies(IEnumerable<LibraryDependency> dependencies,
 
                 if (group != null)
                 {
+                    if (maccatalystFallback != null)
+                    {
+                        object tfmObj;
+                        if (group.Properties.TryGetValue("tfm", out tfmObj))
+                        {
+                            var tfm = (NuGetFramework)tfmObj;
+                            if (tfm.Framework.Equals("xamarin.ios", StringComparison.OrdinalIgnoreCase))
```

Comment:
Can you please use the constants, instead https://github.com/NuGet/NuGet.Client/blob/d2e800b88d6b941732272c3a8cf7115836e98e22/src/NuGet.Core/NuGet.Frameworks/FrameworkConstants.cs#L61.